client/blocks/term-tree-selector/terms.jsx

```diff
@@ -64,28 +60,26 @@ const TermTreeSelectorList = createReactClass( {
 		onChange: PropTypes.func,
 		isError: PropTypes.bool,
 		height: PropTypes.number,
-	},
-
-	getInitialState() {
-		// getInitialState is also used to reset state when a the taxonomy prop changes
-		return {
-			searchTerm: '',
-			requestedPages: [ 1 ],
-		};
-	},
-
-	getDefaultProps() {
-		return {
-			analyticsPrefix: 'Category Selector',
-			searchThreshold: 8,
-			loading: true,
-			terms: [],
-			onSearch: () => {},
-			onChange: () => {},
-			onNextPage: () => {},
-			height: 300,
-		};
-	},
+	};
+
+	static defaultProps = {
+		analyticsPrefix: 'Category Selector',
+		searchThreshold: 8,
+		loading: true,
+		terms: [],
+		onSearch: () => {},
+		onChange: () => {},
+		onNextPage: () => {},
+		height: 300,
+	};
+
+	// initialState is also used to reset state when a the taxonomy prop changes
+	static initialState = {
+		searchTerm: '',
+		requestedPages: [ 1 ],
```

Comment:
The operations on `this.state.requestedPages` seem to be safe, but with this change I'd advocate using `Object.freeze` on any array/object in `Component.defaultProps` or `Component.initialState`.

Reply:
fbbb1bf